src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/loptimizer/heuristics/utils/Var_Analyzer.java

```diff
@@ -0,0 +1,35 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.loptimizer.heuristics.utils;
+
+import org.apache.jena.graph.Node;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Var_Analyzer {
+
+//    varsTotal: a set of all variables that have been calculated
+//    Add all bound variables to bindings
+    public static int calculateVars( final List<Node> vars, final Set<Node> varsTotal, final Set<Node> bindings ) {
```

Comment:
It may be very difficult to understand the code that is using this function because the function has the side effect of modifying both `varsTotal` and `bindings`. It would be better to move this code into the place where you call this function.